### DISCUSSION TITLE
Keep favorites backed up

### DISCUSSION THREAD

**Discussion by aniqueta** [immich-app/immich#1669] [Feature Request]
### Feature detail

It’s awesome to see how fast this project is maturing. 

I’m wondering if it’s possible to for Immich to one day keep track of changes to the ‘favorite’ status of photos on mobile devices, even if those photos are already backed up. I believe Google Photos rolled this out recently for iOS devices, so I suppose it’s technically feasible. Even better is if this can be synced to Apple Photos library on a Mac. Then this could truly be a private replacement for iCloud Photos. 

Thanks for considering. 

### Platform

Mobile App

**Comment by fivestones** [comment]
This would be an amazing feature

**Comment by 2600box** [comment]
I think this would be a key feature to replace iCloud. Hopefully it is possible?

**Comment by fivestones** [comment]
I think along the same lines, this should update the database for photos that have other changes made on iPhone after they have been backed up, like editing a photo or manually changing the location of a photo.

**Comment by sebasfavaron** [comment]
Love this project! This would be key to replacing my current backup solution

**Comment by Juan-Esc** [comment]
+1 for this feature request, would love to see it

**Comment by timonrieger** [comment]
Commenting here to gather some implementation details. Do you expect it to be unidirectional sync from device to immich or bidirectional in both ways? I don’t understand what you mean with the mac library part, could you elaborate please?

**Reply by SimJoSt** [reply]
Uni-directional would be my expectation.

Most people are not aware of the separation between different scopes. A user might not even be aware that they marked a media asset as favorite one time in the Immich app and another time in the phone's gallery.

I would love to backup all my photos, then go through them on the computer / web interface, marking them as favorites and
sometimes I might go through my already backed up media assets on the phone during a long trip and mark them there.
It would be great if those were not separate.
(Yes, I can already do that by only using the Immich app on my phone to mark them as favorites.)

It is also relevant in for using the "Free up space" feature on the phone. Which favorite status is considered?
I've also asked this in: https://github.com/immich-app/immich/discussions/26761
And just realized, the answer is "Both": https://github.com/immich-app/immich/pull/26473

**Reply by SimJoSt** [reply]
I know many people who work intensively with the "Favorites" collection on iPhones. While they could also use the Immich app to manage assets themselves, there are other use-cases, where that is not enough.
When accessing photos for sharing them (from a messenger or an app like LocalSend) the OS asset picker allows for filtering the media by collection, including a favorites one.
If the remote favorite status from Immich is not reflected there, it would break that workflow.

**Comment by aniqueta** [comment]
Yes, a favorite status changed in the photos app on an iPhone should ideally sync with the immich server. (These statuses could change before or after the photo has been transferred to immich.) Additionally if the status changed in the immich server, it should sync back to the iPhone photos library. 

The Mac part was related to a hope that photos synced to immich from an iPhone photos library would also be (one day?) synced to an macOS photos library. Basically imitating iCloud. In that scenario, the favorites status should sync across the iPhone, Mac, and immich server.  I don't think immich evolved to have such cross-device syncing (yet).

**Reply by timonrieger** [reply]
Okay understood. The second part of your message is far out of scope for this feature, so I'd keep it out of the implementation. 

@alextran1502 regarding https://github.com/immich-app/immich/discussions/1669#discussioncomment-5558561 and

> Yes, a favorite status changed in the photos app on an iPhone should ideally sync with the immich server. (These statuses could change before or after the photo has been transferred to immich.) Additionally if the status changed in the immich server, it should sync back to the iPhone photos library. (https://github.com/immich-app/immich/discussions/1669#discussioncomment-15401610)

Could you give some guidance on whether we want bidirectional sync? or should we start with one way sync from device to immich first. I think starting with one-way sync of just the "favorite" status would be the way to go, keeping the scope of the feature small and making sure the implementation is solid in the first place before adding more "syncable" metadata to be supported (location changes, etc.)

**Reply by alextran1502** [reply]
One-way sync is good. The technical challenge is how to propagate the event back to the server. We have two separate sync pipelines: local and remote. I imagine it will happen after both tasks have finished

**Comment by lopcode** [comment]
Glad to see this feature is on your radar - I think it's part of some confusing behaviour I saw after setting Immich up. I originally synced my "Favourites" library on iOS which correctly made them all show as "favourited" as expected. Then I switched to syncing "Recents" to move everything, and now photos sync **before** I have a chance to favourite them via iCloud, meaning the favourite state in Immich is out of sync.

The graduation path seems like it's a common one for users - first sync a subset of my photos, and then enjoy the product and decide to sync them all.

It would be great to have "favourite" status sync more smartly - one-way from iCloud Photos -> Immich would be fine for my use case 🙏
